app/models/user.rb

```diff
@@ -21,8 +23,19 @@ class User < ApplicationRecord
   
   validate :validate_roles
   
-  before_create :generate_reset_token
-  
+  before_create :generate_reset_token, :create_activation_digest
+  def User.digest(string)
```

Comment:
I would use `def self.digest(string)` here. I have never seen this syntax before.